packages/preact-iso/router.js

```diff
@@ -82,15 +104,18 @@ export function Router(props) {
 		} else commit();
 	}, [url]);
 
-	const children = [].concat(...props.children);
-
-	let a = children.filter(c => c.props.path === path);
-
-	if (a.length == 0) a = children.filter(c => c.props.default);
-
-	curChildren.current = a.map((p, i) => cloneElement(p, { path, query }));
+	let p, d, m;
+	[].concat(props.children || []).some(vnode => {
+		const matches = exec(path, vnode.props.path, m = { path, query });
+		if (matches) {
+			return p = cloneElement(vnode, { ...m, ...matches })
+		} else {
+			if (vnode.props.default) d = cloneElement(vnode, m);
+			return undefined;
+		}
+	});
```

Comment:
A function returns `undefined` by default and is only needed if we want to bail out early. Here we don't have code after this statement, so we could rewrite this section as:

```suggestion
	[].concat(props.children || []).some(vnode => {
		const matches = exec(path, vnode.props.path, m = { path, query });
		if (matches) {
			return p = cloneElement(vnode, { ...m, ...matches })
		}
		
		if (vnode.props.default) d = cloneElement(vnode, m);
	});
```